R/log_shiny_input_changes.R

```diff
@@ -0,0 +1,67 @@
+#' Auto logging input changes in Shiny app
+#'
+#' This is to be called in the \code{server} section of the Shiny app.
+#'
+#' Function having very similar behavior as [logger::log_shiny_input_changes()] but adjusted for `teal` needs.
+#'
+#' @param input passed from Shiny \code{server}
+#' @param excluded_inputs character vector of input names to exclude from logging
+#' @param excluded_patterns character of length one including a grep pattern of names to be excluded from logging
+#' @param namespace the name of the namespace
+#' @examples
+#' \dontrun{
+#' library(shiny)
+#'
+#' ui <- bootstrapPage(
+#'   numericInput("mean1", "mean1", 0),
+#'   numericInput("mean2", "mean2", 0),
+#'   numericInput("sd", "sd", 1),
+#'   textInput("title", "title", "title"),
+#'   textInput("foo", "This is not used at all, still gets logged", "foo"),
+#'   passwordInput("password", "Password not to be logged", "secret"),
+#'   plotOutput("plot")
+#' )
+#'
+#' server <- function(input, output) {
+#'   log_shiny_input_changes(input, excluded_inputs = "password", exclude_patterns = "mean")
+#'
+#'   output$plot <- renderPlot({
+#'     hist(rnorm(1e3, input$mean, input$sd), main = input$title)
+#'   })
+#' }
+#'
+#' shinyApp(ui = ui, server = server)
+#' }
+#' @export
+log_shiny_input_changes <- function(
+    input,
+    namespace = NA_character_,
+    excluded_inputs = character(),
+    excluded_patterns = "_width$") {
+  session <- shiny::getDefaultReactiveDomain()
```

Comment:
some odd tabulation
```suggestion
log_shiny_input_changes <- function(
  input,
  namespace = NA_character_,
  excluded_inputs = character(),
  excluded_patterns = "_width$"
) {
  session <- shiny::getDefaultReactiveDomain()
```

Reply:
this must have been styler::style_pkg :D